x-pack/legacy/plugins/index_lifecycle_management/shim.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+
+export interface CoreSetup {
+  server: Legacy.Server;
+}
```

Comment:
You may want to use the [`LegacySetup` pattern](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#introduce-new-plugin-definition-shim) that was recently added to the Migration Guide.

The gist for this is, add a 3rd argument to your plugin lifecycle method that represents all the legacy dependencies, and use the real CoreSetup interface as the first argument. This makes it easier to identify which parts still need to be migrated away from Legacy APIs and which parts are using the real CoreSetup APIs.

Reply:
Thanks, Josh. I attempted to implement this pattern in https://github.com/elastic/kibana/pull/56068/commits/f35f1b58181d5b972af13c5387a3b8fe80ad8e6a, but I'm not entirely sure if this is correct.